From 6d4667870ae7938e329e203e8ca2fbd3f4c5269c Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Wed, 1 Mar 2023 17:30:39 +0100
Subject: [PATCH 5/6] ENH: Enable more selective installation of python files

This patch replaces the whole-directory installation mechanism for
python files by a mechanism that selectively install the scripts and the
resources only. This targets SystoleOS/gentoo-overlay#75
---
 CMake/ctkMacroCompilePythonScript.cmake | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/CMake/ctkMacroCompilePythonScript.cmake b/CMake/ctkMacroCompilePythonScript.cmake
index 4ecf5546..1945d06e 100644
--- a/CMake/ctkMacroCompilePythonScript.cmake
+++ b/CMake/ctkMacroCompilePythonScript.cmake
@@ -93,10 +93,15 @@ macro(ctkMacroCompilePythonScript)
     set(MY_DIRECTORY_TO_INSTALL ${MY_DESTINATION_DIR}/)
   endif()
 
-  # Install python module / resources directory
-  install(DIRECTORY "${MY_DIRECTORY_TO_INSTALL}"
-    DESTINATION "${MY_INSTALL_DIR}" COMPONENT RuntimeLibraries
-    USE_SOURCE_PERMISSIONS)
+  if(MY_SCRIPTS)
+    install(FILES ${MY_SCRIPTS}
+      DESTINATION "${MY_INSTALL_DIR}" COMPONENT RuntimeLibraries)
+  endif()
+
+  if(MY_RESOURCES)
+    install(FILES ${MY_RESOURCES}
+      DESTINATION "${MY_INSTALL_DIR}" COMPONENT RuntimeLibraries)
+  endif()
 
   if(NOT MY_GLOBAL_TARGET)
     ctkFunctionAddCompilePythonScriptTargets(${target})
-- 
2.39.2

